src/core/chart.js

```diff
@@ -0,0 +1,252 @@
+import _ from 'underscore';
+
+
+var PRECISION = 2;
+
+/* This function takes one or more JSON objects from the 
+ * "timeseries" API call with this format:
+ * 
+ * {
+ * "id": "tasmax_mClim_BCCAQv2_bcc-csm1-1-m_historical-rcp45_r1i1p1_20700101-20991231_Canada",
+ * "units": "degC",
+ * "data": {
+ *   "2085-01-15T00:00:00Z": -17.498223073165622,
+ *   "2085-02-15T00:00:00Z": -15.54878007851129,
+ *   "2085-03-15T00:00:00Z": -11.671093808333737,
+ *                    ...
+ *    }
+ * }
+ * 
+ * and return a C3 graph object displaying them.
+ * 
+ * It takes an arbitrary number of data objects, but no more than
+ * two separate unit types. Allowable data resolutions are monthly(12), 
+ * seasonal (4), or yearly (1); an error will be thrown 
+ * if this function is called on data with another time resolution.
+ */
+var timeseriesToAnnualCycleGraph = function(metadata, ...data) {
+
+  //blank graph data object to be populated - holds data values
+  //and individual-timeseries-level display options.
+  var c3Data = {
+      columns: [],
+      types: {},
+      labels: {},
+      axes: {}
+  };
+
+  var yUnits = "";
+  var y2Units = "";
+  
+  var getTimeseriesName = shortestUniqueTimeseriesNamingFunction(metadata, data);
+  
+  //Add each timeseries to the graph
+  for(var i = 0; i < data.length; i++) {
+
+    //get metadata for this timeseries
+    var timeseries = data[i];
+    var timeseriesMetadata = _.find(metadata, function(m) {return m.unique_id === timeseries.id;});  
+    var timeseriesName = getTimeseriesName(timeseriesMetadata);
+       
+    //add the actual data to the graph
+    c3Data.columns.push([timeseriesName].concat(getMonthlyData(timeseries.data, timeseriesMetadata.timescale)));
+    
+    //monthly data is displayed as a line graph, but yearly and seasonal
+    //display as step graphs.
+    c3Data.types[timeseriesName] = timeseriesMetadata.timescale == "monthly" ? "line" : "step";
+
+    //Each timeseries needs to be associated with a y-axis.
+    //Two different variables measured with the same units, 
+    //like tasmin (degrees C) and tasmax (degrees C)
+    //can share a y-axis. 
+    //Variables with different units require seperate axes.
+    //C3 can theoretically support indefinite numbers of y-axes,
+    //but that would be hard for a user to make sense of, 
+    //so it's capped at two here.
+    if((!yUnits) || timeseries.units == yUnits) {
+      yUnits = timeseries.units;
+      c3Data.axes[timeseriesName] = "y";
+    }
+    else if((!y2Units) || timeseries.units == y2Units) {
+      y2Units = timeseries.units;
+      c3Data.axes[timeseriesName] = "y2";
+    }
+    else {
+      throw new Error("Error: too many data axes required for graph");
```

Comment:
Also, nitpick: No need to repeat the word "Error" in the string in the Error object.